homeassistant/components/wiz/light.py

```diff
@@ -0,0 +1,348 @@
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+        except (ConnectionRefusedError, TimeoutError, WizLightTimeOutError) as ex:
+            _LOGGER.debug(ex)
+            self.update_state_unavailable()
+        else:
+            if self._light.state is None:
+                self.update_state_unavailable()
+            else:
+                self.update_state_available()
+        _LOGGER.debug(
+            "[wizlight %s] updated state: %s and available: %s",
+            self._light.ip,
+            self._state,
+            self._available,
+        )
+
+    def update_brightness(self):
+        """Update the brightness."""
+        if self._light.state.get_brightness() is None:
+            return
+        brightness = self._light.state.get_brightness()
+        if 0 <= int(brightness) <= 255:
+            self._brightness = int(brightness)
+        else:
+            _LOGGER.error(
+                "Received invalid brightness : %s. Expected: 0-255", brightness
+            )
+            self._brightness = None
+
+    def update_temperature(self):
+        """Update the temperature."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is None or colortemp == 0:
+            self._temperature = None
+            return
+
+        _LOGGER.debug(
+            "[wizlight %s] kelvin from the bulb: %s", self._light.ip, colortemp
+        )
+        temperature = color_utils.color_temperature_kelvin_to_mired(colortemp)
+        self._temperature = temperature
+
+    def update_color(self):
+        """Update the hs color."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is not None and colortemp != 0:
+            self._hscolor = None
+            return
+        if self._light.state.get_rgb() is None:
+            return
+
+        rgb = self._light.state.get_rgb()
+        if rgb[0] is None:
+            # this is the case if the temperature was changed
+            # do nothing until the RGB color was changed
+            return
+        warmwhite = self._light.state.get_warm_white()
+        if warmwhite is None:
+            return
+        self._hscolor = convertHSfromRGBCW(rgb, warmwhite)
+
+    def update_effect(self):
+        """Update the bulb scene."""
+        self._effect = self._light.state.get_scene()
+
+    async def get_bulb_type(self):
+        """Get the bulb type."""
+        if self._bulbtype is not None:
+            return self._bulbtype
+        try:
+            self._bulbtype = await self._light.get_bulbtype()
+            _LOGGER.info(
```

Comment:
Please move the log out of the try except block.

Reply:
already removed